nav2_rrtconnect_planner/src/rrt_connect.cpp

```diff
@@ -0,0 +1,269 @@
+
+  ompl_initialized_ = true;
+}
+
+nav_msgs::msg::Path RRTConnect::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  if (!ompl_initialized_) {
+    RCLCPP_INFO(node_->get_logger(), "Initializing OMPL!");
+    initializeOMPL();
+  }
+  // Returning variable
+  nav_msgs::msg::Path global_path;
+
+  // Checking if the goal and start state is in the global frame
+  if (start.header.frame_id != global_frame_) {
+    RCLCPP_ERROR(
+      node_->get_logger(), "Planner will only except start position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  if (goal.header.frame_id != global_frame_) {
+    RCLCPP_INFO(
+      node_->get_logger(), "Planner will only except goal position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  ompl::base::ScopedState<> ss_start(ompl_state_space_), ss_goal(ompl_state_space_);
+  ss_start = std::vector<double>{start.pose.position.x, start.pose.position.y};
+  ss_goal = std::vector<double>{goal.pose.position.x, goal.pose.position.y};
+  // Setting state and goal state
+  ss_->setStartAndGoalStates(ss_start, ss_goal);
+  // Assigning RRTConnect
+  ss_->setPlanner(
+    ompl::base::PlannerPtr(
+      new ompl::geometric::RRTConnect(
+        ss_->getSpaceInformation())));
+  // Load settings
+  ss_->setup();
+  // Counter for replanning
+  int counter = 0;
+  // Planning
+  while (!ss_->solve(solve_time_) && counter < max_trials_) {
+    ++counter;
+  }
+  // Plan found
+  if (counter < max_trials_) {
+    RCLCPP_INFO(
+      node_->get_logger(), "Found %d solutions",
+      ss_->getProblemDefinition()->getSolutionCount());
+    // Path simplification
+    ss_->simplifySolution();
+    auto solution_path = ss_->getSolutionPath();
+
+    global_path.poses.clear();
+    global_path.header.stamp = node_->now();
+    global_path.header.frame_id = global_frame_;
+    // Increasing number of path points
+    int min_num_states = round(solution_path.length() / costmap_->getResolution());
+    solution_path.interpolate(min_num_states);
+
+    global_path.poses.reserve(solution_path.getStates().size());
+    for (const auto ptr : solution_path.getStates()) {
+      global_path.poses.emplace_back(getPoseStampedFromState(*ptr));
+    }
+    global_path.poses[global_path.poses.size() - 1] = goal;
+  }
+
+  return global_path;
+}
+
+bool RRTConnect::isStateValid(const ompl::base::State * state)
+{
+  if (!ss_->getSpaceInformation()->satisfiesBounds(state)) {
+    return false;
+  }
+
+  ompl::base::ScopedState<> ss(ompl_state_space_);
+  ss = state;
+  unsigned int mx, my;
+  costmap_->worldToMap(ss[0], ss[1], mx, my);
+
+  auto cost = costmap_->getCost(mx, my);
+  if (cost == nav2_costmap_2d::LETHAL_OBSTACLE ||
+    cost == nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE ||
+    ( !allow_unknown_ && cost == nav2_costmap_2d::NO_INFORMATION))
+  {
```

Comment:
With the way this state validation check is done, it seems that using an SE2 space adds an extra rotational dimension that will be searched in without any benefit